### PR TITLE
Modernize the project for Composer 2, PHP 8, Drupal 10 among others

### PR DESCRIPTION
**Pull request by m4olivei**
Goal here is to update what is necessary to allow to install and run in environments with Composer 2, PHP 8 and Drupal 10, among others. One of the primary considerations for our purposes is the [major version bump of the Composer plugin API](https://getcomposer.org/upgrade/UPGRADE-2.0.md). That version bump means the following is needed:

* There are a few dependencies where at the currently required version, don't support Composer plugin API v2, including:
  * phpro/grumphp
  * dealerdirect/phpcodesniffer-composer-installer
  * localheinz/composer-normalize
* Bumping `phpro/grumphp` means we have to make some adjustments to the `grumphp.yml` config file. See the following for details: https://github.com/phpro/grumphp/releases/tag/v0.19.0
* API updates to the composer plugin itself

Other updates that we need on account of testing on PHP 8 environments as well as testing Drupal 10 projects:

* Bumping phpunit/phpunit to 9.5
* Bumping Drupal to Drupal 10
* Bumping PHP to version 8

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,48 +1,58 @@
...
+    "composer-plugin-api": "^1.1 || ^2.0",
+    "dealerdirect/phpcodesniffer-composer-installer": "^0.5.0 || ^1.0",
     "drupal/coder": "^8.2",
     "jakub-onderka/php-parallel-lint": "^1.0",
-    "localheinz/composer-normalize": "^1.3",
+    "localheinz/composer-normalize": "^2.4",
     "nikic/php-parser": "^4.2",
     "phpmd/phpmd": "^2",
     "phpmetrics/phpmetrics": "^2.4",
-    "phpro/grumphp": "^0.15.2",
```
Comment (e0ipso) [inline composer.json:27]: We should look into using the phar instead.

https://github.com/phpro/grumphp#installation

**Comment by m4olivei** [comment]
I'm realizing that for our particular needs, we also need to have this be PHP 8 compatible. Just making it Composer 2 compatible isn't quite enough.

I'm going to next layer on some PHP 8 compatibility things. I might split that out, but seeing as the repo has been rather quiet, I'm thinking it may be OK to carry on here in this PR / branch rather than splitting it out ... b/c lazy. Happy to be prodded into splitting it up though.

File: composer.json [hunk 2]
```diff
@@ -1,48 +1,54 @@
...
   "require": {
-    "php": ">=7.3",
-    "composer-plugin-api": "^1.1",
-    "dealerdirect/phpcodesniffer-composer-installer": "^0.5.0",
+    "php": ">=8.1",
+    "composer-plugin-api": "^1.1 || ^2.0",
+    "dealerdirect/phpcodesniffer-composer-installer": "^0.5.0 || ^1.0",
     "drupal/coder": "^8.2",
-    "jakub-onderka/php-parallel-lint": "^1.0",
-    "localheinz/composer-normalize": "^1.3",
```
Comment (e0ipso) [inline composer.json]: I am also happy to leave old dependencies behind. As soon as we start adding `OR` the dependency matrix becomes bigger and bigger.

How about we target this for a major version?

File: grumphp.yml [hunk 3]
```diff
@@ -1,13 +1,8 @@
-parameters:
-  git_dir: .
-  bin_dir: vendor/bin
+grumphp:
```
Comment (e0ipso) [inline grumphp.yml:1]: I recently worked on a grumphp setup. Maybe https://github.com/Lullabot/parse.ly-php/blob/main/grumphp.yml can be of inspiration.

My goal is to modernize the setup, as I know some of the tools here have faded a bit in these past years.

**Comment by e0ipso** [comment]
I love this modernization. Sad that Travis is not as popular as it once was, but glad that it's useful for you.

File: grumphp.yml (hunk 3 above)
Comment (m4olivei) [reply grumphp.yml:1]: Thanks for the reference, I've updated it accordingly, wasn't too far off. Left in the additional tools for now.

File: composer.json (hunk 1 above)
Comment (m4olivei) [reply composer.json:27]: Done!

File: composer.json (hunk 2 above)
Comment (m4olivei) [reply composer.json]: Sounds good! 2.0 here we come.

**Comment by m4olivei** [comment]
@e0ipso humm, I just realized that [semantic-release-conventional-commits](https://www.npmjs.com/package/semantic-release-conventional-commits) is going to run when we merge this, and I haven't been diligent abour marking breaking changes in any of the commits. What do you recommend? Should I interactive rebase and mark the appropriate commits as breaking?

File: src/Composer/TravisDrupalModulePlugin.php [hunk 4]
```diff
@@ -78,25 +78,41 @@ final class TravisDrupalModulePlugin implements
      * Apply plugin modifications to Composer
      *
      * @param Composer    $composer The composer API.
-     * @param IOInterface $theIo    IO interface to log to the terminal.
+     * @param IOInterface $io       IO interface to log to the terminal.
```
Comment (e0ipso) [inline src/Composer/TravisDrupalModulePlugin.php:81]: This used to fail PHP EA validatio because it was too short. Glad to see that nonsense gone.

**Comment by e0ipso** [comment]
@m4olivei I think you can use "Squash and merge" so only one commit is created. There you can add the commit message and the `BREAKING CHANGE` keyword that will trigger a new major version.

**Review by e0ipso — APPROVED** [review]
:100: :shipit:

File: src/Composer/TravisDrupalModulePlugin.php (hunk 4 above)
Comment (m4olivei) [reply src/Composer/TravisDrupalModulePlugin.php:81]: It's not gone, I forced the commit even with it 😊

**Pull request merged**